rust/lance/src/dataset/write/insert.rs

```diff
@@ -181,6 +183,8 @@ impl<'a> InsertBuilder<'a> {
     ) -> Result<(Transaction, WriteContext<'_>)> {
         let mut context = self.resolve_context().await?;
 
+        info!(target: TRACE_DATASET_EVENTS, event=DATASET_WRITING_EVENT, path=context.base_path.to_string(), mode=?context.params.mode);
```

Comment:
What about here? Because there are two `execute` methods. And they shared the same preparation methods.